tests/test_base_track_target_and_take_image.py

```diff
@@ -0,0 +1,134 @@
+        self.instrument_name = "GenericCam"
+
+    @property
+    def tcs(self):
+        return self.mtcs
+
+    @property
+    def camera(self):
+        return self.lsstcam
+
+    @classmethod
+    def get_schema(cls):
+        url = "https://github.com/"
+        path = "lsst-ts/ts_standardscripts/tree/develop/tests/test_base_track_target_and_take_image.yaml"
+        schema_dict = cls.get_base_schema()
+        schema_dict["$id"] = f"{url}{path}"
+        schema_dict["title"] = "GenericTrackTargetAndTakeImage v1"
+        schema_dict["description"] = "Configuration for GenericTargetAndTakeImage."
+
+        return schema_dict
+
+    async def load_playlist(self):
+        """Load playlist."""
+        raise NotImplementedError()
+
+    async def assert_feasibility(self):
+        """Verify that the system is in a feasible state to execute the
+        script.
+        """
+        raise NotImplementedError()
+
+    async def track_target_and_setup_instrument(self):
+        """slewing of telescope and setting up of instrument.
+
+        Ideally this would be done in parallel to save time.
+        """
+        raise NotImplementedError()
+
+    async def take_data(self):
+        """Implement method to take data."""
+        raise NotImplementedError()
+
+    async def stop_tracking(self):
+        """Implement method to stop tracking."""
+        raise NotImplementedError()
+
+    def get_instrument_name(self):
+        """Get instrument name.
+
+        Returns
+        -------
+        instrument_name: `string`
+        """
+        return self.instrument_name
+
+
+class TestBaseTrackTargetAndTakeImage(
+    standardscripts.BaseScriptTestCase, unittest.IsolatedAsyncioTestCase
+):
+    """Test BaseTrackTargetAndTakeImage using the DummyBlockScript script."""
+
+    async def basic_make_script(self, index):
+        self.script = GenericTrackTargetAndTakeImage(index=index)
+        self.script.mtcs = unittest.mock.AsyncMock()
+        self.script.lsstcam = unittest.mock.AsyncMock()
+
+        return (self.script,)
+
+    async def test_configure_ignore(self):
+        configuration_basic_with_ignore = dict(
+            ra="10:00:00",
+            dec="-10:00:00",
+            rot_sky=0.0,
+            name="unit_test_target",
+            obs_time=7.0,
+            num_exp=2,
+            exp_times=[2.0, 1.0],
+            band_filter="r",
+            reason="Unit testing",
+            program="UTEST",
+            note="Note_utest",
+            ignore=["mtm1m3", "mtrotator", "mtoods", "not-comp"],
+        )
+
+        async with self.make_script():
+            await self.configure_script(**configuration_basic_with_ignore)
+            # Verify ignoring components
+            self.script.tcs.disable_checks_for_components.assert_called_once_with(
+                components=configuration_basic_with_ignore["ignore"]
+            )
+            self.script.camera.disable_checks_for_components.assert_called_once_with(
```

Comment:
We should only ignore tcs components, please remove the feature to ignore camera components.